Source/SwiftLintFramework/Rules/Lint/StrongIBOutletRule.swift

```diff
@@ -18,24 +20,44 @@ public struct StrongIBOutletRule: ConfigurationProviderRule, ASTRule, OptInRule,
             "@IBOutlet weak ↓var label: UILabel?",
             "@IBOutlet unowned ↓var label: UILabel!",
             "@IBOutlet weak ↓var textField: UITextField?"
-        ].map(wrapExample)
+        ].map(wrapExample),
+        corrections: [
+            "@IBOutlet weak ↓var label: UILabel?": "@IBOutlet var label: UILabel?",
+            "@IBOutlet unowned ↓var label: UILabel!": "@IBOutlet var label: UILabel!",
+            "@IBOutlet weak ↓var textField: UITextField?": "@IBOutlet var textField: UITextField?"
+        ].reduce(into: [String: String]()) { $0[wrapExample($1.key)] = wrapExample($1.value) }
     )
 
     public func validate(file: SwiftLintFile, kind: SwiftDeclarationKind,
                          dictionary: SourceKittenDictionary) -> [StyleViolation] {
+        return violationRanges(in: file, kind: kind, dictionary: dictionary).map {
+            StyleViolation(ruleDescription: type(of: self).description,
+                           severity: configuration.severity,
+                           location: Location(file: file, characterOffset: $0.location))
+        }
+    }
+
+    public func violationRanges(in file: SwiftLintFile, kind: SwiftDeclarationKind,
+                                dictionary: SourceKittenDictionary) -> [NSRange] {
         guard kind == .varInstance,
             case let attributes = dictionary.enclosedSwiftAttributes,
             attributes.contains(.iboutlet),
             attributes.contains(.weak),
-            let offset = dictionary.offset else {
+            let offset = dictionary.offset,
+            let range = file.stringView.byteRangeToNSRange(start: offset, length: 3) else {
                 return []
         }
+        return [range]
+    }
 
-        return [
-            StyleViolation(ruleDescription: type(of: self).description,
-                           severity: configuration.severity,
-                           location: Location(file: file, byteOffset: offset))
-        ]
+    public func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
+        let prefixString = file.contents.bridge().substring(to: violationRange.location)
```

Comment:
we should use the range of the `weak` attribute that SourceKitten provides instead:

```
{
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 35,
  "key.offset" : 0,
  "key.substructure" : [
    {
      "key.accessibility" : "source.lang.swift.accessibility.internal",
      "key.attributes" : [
        {
          "key.attribute" : "source.decl.attribute.weak",
          "key.length" : 4,
          "key.offset" : 10
        },
        {
          "key.attribute" : "source.decl.attribute.iboutlet",
          "key.length" : 9,
          "key.offset" : 0
        }
      ],
      "key.kind" : "source.lang.swift.decl.var.global",
      "key.length" : 19,
      "key.name" : "label",
      "key.namelength" : 5,
      "key.nameoffset" : 19,
      "key.offset" : 15,
      "key.setter_accessibility" : "source.lang.swift.accessibility.internal",
      "key.typename" : "UILabel?"
    }
  ]
}
```

Reply:
Great idea! How can I generate this dictionary for the string in the violation range? I know it's possible to get it for the entire file, but it's not very efficient to go through it and filter for the violation range.

Reply:
You can use `dictionary.swiftAttributes` in `violationRanges` (instead of `enclosedSwiftAttributes`)

Reply:
I was referring to `substitution(for:in:)`. The only inputs are NSRange and SwiftLintFile, but no dictionary. I can call `file.structureDictionary`, but this gives me the dictionary for the entire file and not just for the violation range. I figured there must be a better way than to traverse this dictionary to find a declaration that falls within the violation range.

Reply:
To make things easier, I'd change the violation location to be in the `weak`/`unowned` keyword instead, this way you can keep most of the logic in `violationRanges`. That's the main use case that `SubstitutionCorrectableASTRule` was created for.